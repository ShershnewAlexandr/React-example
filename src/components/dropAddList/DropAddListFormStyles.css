.add-button {
    margin-top: 20px;
    width: 100px;
    height: 24px;
    border-radius: 30px;
    border: 2px blue solid;
    font: 12px Roboto;
    vertical-align: center;
    color: blue;
    transition: all ease-out 0.2s;
    cursor: pointer;
    outline: none;
    background-color: white;
}

.add-button:hover {
    color: white;
    background-color: blue;
}

.drop-button {
    margin-top: 20px;
    width: 100px;
    height: 25px;
    border-radius: 30px;
    border: 2px blue solid;
    font: 13px Roboto;
    vertical-align: center;
    color: blue;
    transition: all ease-out 0.2s;
    cursor: pointer;
    outline: none;
    background-color: white;
    font-size: 12px;
    z-index: 10;
}

.drop-button:hover {
    color: white;
    background-color: blue;
}

.drop-icon{
    font-size: 15px;
    margin-right: 6px;
}

.plus-icon{
    font-size: 19px !important;
    margin-top: 0;
    margin-right: 4px;
}

.drop-add-list-hr{
    margin-top: 40px;
    margin-bottom: 15px;
    width: 100%;
}