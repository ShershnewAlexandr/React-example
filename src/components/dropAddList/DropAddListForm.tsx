import React from 'react';
import { WrappedFieldArrayProps } from 'redux-form';
import { Grid } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import './DropAddListFormStyles.css';

interface DropAddListFormProps {
  maxElements: number;
  formRender: React.FC<any>;
  disabled: boolean;
}

export function DropAddListForm({
  maxElements,
  formRender,
  fields,
  disabled,
}: DropAddListFormProps & WrappedFieldArrayProps<undefined>) {
  const { t } = useTranslation();
  const addClick = () => {
    if (fields.length < maxElements) fields.push(undefined);
  };

  const elementsForm = fields.map((element: string, index: number) => (
    <React.Fragment key={index}>
      <Grid container direction="row" justify="flex-end">
        <button
          type="button"
          className={disabled ? 'drop-button disabled' : 'drop-button'}
          onClick={() => fields.remove(index)}
          disabled={disabled}
        >
          <Grid container justify="center" alignItems="center">
            <Grid item>
              <i className="material-icons drop-icon">clear</i>
            </Grid>
            <Grid item>{t('dropAddList.drop')}</Grid>
          </Grid>
        </button>
      </Grid>
      {formRender({ element, disabled, t })}
      {index !== fields.length - 1 && <hr className="drop-add-list-hr" />}
    </React.Fragment>
  ));

  return (
    <>
      <Grid
        container
        direction="column"
        justify="flex-start"
        alignItems="flex-end"
      >
        {elementsForm}
      </Grid>
      <Grid container justify="flex-start">
        <Grid item>
          <button
            className={disabled ? 'add-button disabled' : 'add-button'}
            type="button"
            onClick={addClick}
            disabled={disabled}
          >
            <Grid container justify="center" alignItems="center">
              <Grid item>
                <i className="material-icons md-24 plus-icon">add</i>
              </Grid>
              <Grid item>{t('dropAddList.add')}</Grid>
            </Grid>
          </button>
        </Grid>
      </Grid>
    </>
  );
}

export default DropAddListForm;
