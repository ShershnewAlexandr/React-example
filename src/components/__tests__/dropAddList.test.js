import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import '../../setupTests';
import { DropAddListForm } from '../dropAddList/DropAddListForm';

jest.mock('i18next', () => ({
  use: function() {
    return this;
  },
  init: jest.fn(),
}));
jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: str => str,
  }),
}));

describe('dropAddListForm', () => {
  const props = {
    maxElements: 5,
    formRender: jest.fn(),
    fields: [1, 2, 3],
    disabled: false,
    t: jest.fn(),
  };

  const dropAllListForm = shallow(<DropAddListForm {...props} />);

  it('snapshot', () => {
    expect(toJson(dropAllListForm)).toMatchSnapshot();
  });

  it('fields number', () => {
    expect(dropAllListForm.find('.drop-icon')).toHaveLength(3);
  });
});
