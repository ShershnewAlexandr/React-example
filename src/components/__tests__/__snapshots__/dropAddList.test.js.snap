// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropAddListForm snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Grid))
    alignItems="flex-end"
    container={true}
    direction="column"
    justify="flex-start"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      direction="row"
      justify="flex-end"
    >
      <button
        className="drop-button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <WithStyles(ForwardRef(Grid))
          alignItems="center"
          container={true}
          justify="center"
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <i
              className="material-icons drop-icon"
            >
              clear
            </i>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            dropAddList.drop
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </button>
    </WithStyles(ForwardRef(Grid))>
    <hr
      className="drop-add-list-hr"
    />
    <WithStyles(ForwardRef(Grid))
      container={true}
      direction="row"
      justify="flex-end"
    >
      <button
        className="drop-button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <WithStyles(ForwardRef(Grid))
          alignItems="center"
          container={true}
          justify="center"
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <i
              className="material-icons drop-icon"
            >
              clear
            </i>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            dropAddList.drop
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </button>
    </WithStyles(ForwardRef(Grid))>
    <hr
      className="drop-add-list-hr"
    />
    <WithStyles(ForwardRef(Grid))
      container={true}
      direction="row"
      justify="flex-end"
    >
      <button
        className="drop-button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <WithStyles(ForwardRef(Grid))
          alignItems="center"
          container={true}
          justify="center"
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <i
              className="material-icons drop-icon"
            >
              clear
            </i>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            dropAddList.drop
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </button>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    justify="flex-start"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <button
        className="add-button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <WithStyles(ForwardRef(Grid))
          alignItems="center"
          container={true}
          justify="center"
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <i
              className="material-icons md-24 plus-icon"
            >
              add
            </i>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            dropAddList.add
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </button>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;
